### PR TITLE
Improve unit test verification

### PR DESCRIPTION
**Pull request by stuartpa**
This PR is being merged as part of https://github.com/microsoft/go-sqlcmd/pull/205

### CODE REVIEW

File: cmd/modern/root/query.go [hunk 1]
```diff
@@ -63,11 +63,12 @@ func (c *Query) run() {
 		line = console.NewConsole("")
 		defer line.Close()
 	}
-	s := mssql.Connect(endpoint, user, line)
+	sql := mssql.New(false)
```
Comment (shueybubbles) [inline cmd/modern/root/query.go:66]: >ew(false) [](http://example.com/codeflow?start=15&length=9)

Boolean parameters are where I miss named parameters in C#. I have to look up the New function to figure out what is being set to false.

File: internal/mssql/factory.go [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package mssql
+
+func New(unitTesting bool) MssqlInterface {
```
Comment (shueybubbles) [inline internal/mssql/factory.go:3]: >fun [](http://example.com/codeflow?start=0&length=3)

it's a little weird to me for the production code and the test code to live in the same binary.

Could the mock implementation only live in a _test.go file and the factory accept an optional delegate used to create the instance? If no delegate is passed, it uses the default production type. 
The unit tests would pass a delegate that creates the mock.
Alternatively we could use a build tag to control inclusion of the mock type and make sure the official product build sets the tag to exclude it.

File: cmd/modern/root/query.go (hunk 1 above)
Comment (stuartpa) [reply cmd/modern/root/query.go:66]: Good point, in my IDE (Goland), I see:

	sql := mssql.New( unitTesting: false)
	
The IDE puts the variable name in the display.  But for code reviews and other IDEs that don't do this, it is not clear at all.

I'll move this to the Options pattern which is idiomatic for golang

Comment (stuartpa) [reply cmd/modern/root/query.go:66]: Moved to an Options parameter in:

[https://github.com/microsoft/go-sqlcmd/pull/205](https://github.com/microsoft/go-sqlcmd/pull/205)

File: internal/mssql/factory.go (hunk 2 above)
Comment (stuartpa) [reply internal/mssql/factory.go:3]: I've made a note to talk about this at our next sync.  I'm doing this to speed up unit testing (of other units), I think the mock impl needs to be available in prod code here, but let's chat

**Pull request closed without merging**